src/lib/snack-bar/snack-bar-container.ts

```diff
@@ -94,11 +94,11 @@ export class MatSnackBarContainer extends BasePortalOutlet implements OnDestroy
   onAnimationEnd(event: AnimationEvent) {
     const {fromState, toState} = event;
 
-    if ((toState === 'void' && fromState !== 'void') || toState.startsWith('hidden')) {
+    if (toState === 'void' && fromState !== 'void') {
```

Comment:
We should not need to check the from state.  If we don't take action in cases where we go to `void` from `void` then it results in leaving the snackbar-container in the DOM indefinitely.

Reply:
This isn't something I introduced in this PR, I just had to rework the condition. If I remember correctly, the case where we could get the `fromState` and `toState` to both be `void` was on enter, which shouldn't be an issue for this PR.

Reply:
Hmm, in my testing I didn't see any issue with removing the `fromState` check.  I think the difference now is that we are using `void` instead of `hidden` for our hiding state.

Reply:
I'm inclined to keep it just because it shouldn't hurt anything and these changes are dangerous enough as they are.